sdk/clientcore/core/src/main/java/io/clientcore/core/instrumentation/logging/LoggingEvent.java

```diff
@@ -305,30 +300,72 @@ private String getMessageWithContext(String message) {
                 jsonWriter.writeStringField("message", message);
             }
 
-            if (globalPairs != null) {
-                for (Map.Entry<String, Object> kvp : globalPairs.entrySet()) {
-                    jsonWriter.writeUntypedField(kvp.getKey(), kvp.getValue());
-                }
-            }
+            writeContext(jsonWriter);
 
-            if (keyValuePairs != null) {
-                for (Map.Entry<String, Object> kvp : keyValuePairs.entrySet()) {
-                    jsonWriter.writeUntypedField(kvp.getKey(), kvp.getValue());
-                }
-            }
+            jsonWriter.writeEndObject().flush();
 
-            if (eventName != null) {
-                jsonWriter.writeStringField(EVENT_NAME_KEY, eventName);
-            }
+            return outputStream.toString(StandardCharsets.UTF_8);
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    void recordExceptionDetails(Throwable throwable, String message) {
+        addKeyValueInternal(EXCEPTION_TYPE_KEY, throwable.getClass().getCanonicalName());
+        if (message != null) {
+            addKeyValueInternal(EXCEPTION_MESSAGE_KEY, message);
+        }
+
+        if (logger != null && logger.canLogAtLevel(LogLevel.VERBOSE)) {
+            StringBuilder stackTrace = new StringBuilder();
+            DefaultLogger.appendThrowable(stackTrace, throwable);
```

Comment:
Future enhancement: It's allowable for exceptions to disable stack traces, might want to check if there is a stack trace before adding the key-value pair to the structured log message. And possibly if stack traces were disabled log

`EXCEPTION_STACKTRACE_KEY: <stacktrace disabled by exception>`

to indicate, "sorry, can't help you here"

Reply:
it's interesting. According to throwable docs, when you disable stack trace (disable it being writeable), `getStackTrace` returns zero element array.

The `printStackTrace` through still writes the message like `IOException: connection dropped` which results in us populating `exception.stacktrace` with this value. When stack traces are not recorded, we just don't populate the property.

I don't mind doing something like 

```suggestion
           if (throwable.getStackTrace().length > 0) {
               DefaultLogger.appendThrowable(stackTrace, throwable);
           }
```

but I'm not sure I understand the need?

Reply:
I think that would help, just in case something odd happens